### PR TITLE
[11.0][MIG] sale_blanket_order

### PR DESCRIPTION
**Pull request by AdriaGForgeFlow**
Migration from #595 to v11

### CODE REVIEW

File: sale_blanket_order/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    'license': 'AGPL-3',
+    'author': 'Acsone SA/NV,Odoo Community Association (OCA)',
+    'version': '11.0.1.0.0',
+    'website': 'https://github.com/OCA/sale-workflow',
+    'summary': "Blanket Orders",
+    'depends': [
+        'sale',
+        'web_action_conditionable',
+    ],
+    'data': [
```
Comment (nikul-serpentcs) [inline sale_blanket_order/__manifest__.py:15]: @ageficent `data` sequence should be [this](https://github.com/OCA/maintainer-tools/blob/master/template/module/__manifest__.py#L29)

File: sale_blanket_order/models/sale_orders.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    blanket_order_id = fields.Many2one(
+        'sale.blanket.order', string='Origin blanket order',
+        related='order_line.blanket_line_id.order_id',
+        readonly=True)
+
+
+class SaleOrderLine(models.Model):
+    _inherit = 'sale.order.line'
+
+    blanket_line_id = fields.Many2one('sale.blanket.order.line')
```
Comment (nikul-serpentcs) [inline sale_blanket_order/models/sale_orders.py]: add string also

File: sale_blanket_order/report/templates.xml [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+               <tbody class="sale_tbody">
+                    <t t-foreach="doc.lines_ids" t-as="l">
+                        <tr>
+                            <td><span t-field="l.product_id"/></td>
+                            <td class="text-right">
+                                <span t-field="l.price_unit"/>
+                            </td>
+                            <td class="text-right">
+                                <span t-field="l.original_qty"/>
+                                <!-- <span t-field="l.product_uom" groups="product.group_uom"/> -->
```
Comment (nikul-serpentcs) [inline sale_blanket_order/report/templates.xml]: remove dead code

File: sale_blanket_order/tests/test_blanket_orders.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        # date in the past
+        with self.assertRaises(UserError):
+            blanket_order.action_confirm()
+
+        blanket_order.validity_date = fields.Date.to_string(tomorrow)
+        blanket_order.action_confirm()
+
+        self.assertEqual(blanket_order.state, 'open')
+
+        wizard1 = self.env['sale.blanket.order.wizard'].with_context(
```
Comment (nikul-serpentcs) [inline sale_blanket_order/tests/test_blanket_orders.py]: @ageficent This object `self.env['sale.blanket.order.wizard']` use multiple time, so use as global variable

File: sale_blanket_order/views/sale_config_settings.xml [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). -->
+
+<odoo>
+
+    <record model="ir.ui.view" id="sale_config_settings_form_view">
+        <field name="name">res.config.settings.form (in sale_blanket_order)</field>
+        <field name="model">res.config.settings</field>
+        <field name="inherit_id" ref="sale.res_config_settings_view_form"/>
+        <field name="arch" type="xml">
+            <xpath expr="//div[@class='row mt16 o_settings_container'][position() = 3]" position="after">
```
Comment (nikul-serpentcs) [inline sale_blanket_order/views/sale_config_settings.xml]: use here `hasclass`

**Review by nikul-serpentcs — COMMENTED** [review]
Improve Code

File: sale_blanket_order/__manifest__.py (hunk 1 above)
Comment (AdriaGForgeFlow) [reply sale_blanket_order/__manifest__.py:15]: In order for the module to be installed properly the sequence must be the one currently set so the following line (in blanket_orders.xml line 22) finds the correct reference in the wizard .xml:
<button name="%(action_create_sale_order)d" string="Create Sale Order"

Comment (nikul-serpentcs) [reply sale_blanket_order/__manifest__.py:15]: @ageficent In `security`, `XML` must be First after `CSV`

Comment (AdriaGForgeFlow) [reply sale_blanket_order/__manifest__.py:15]: Oh, OK. I hadn't noticed that. Thanks :+1:

**Comment by marcelofrare** [comment]
When I create sale order, I would like to have the search field on the top of the tree view to find the product in case there are many lines.
If you teach me how to do, I'll do it!
Thanks

**Review by marcelofrare — APPROVED** [review]
I did the functional test on runbot and it's ok for me.

**Review by JordiBForgeFlow — APPROVED** [review]

**Review by LoisRForgeFlow — APPROVED** [review]
Functional test :+1:

**Review by nikul-serpentcs — APPROVED** [review]
Only Code Review LGTM :+1:

**Pull request merged**